api/service/projects_service.go

```diff
@@ -250,3 +283,116 @@ func (service *projectsService) filterAuthorizedProjects(ctx context.Context, pr
+	if service.updateProjectConfig.Endpoint == "" || service.updateProjectConfig.PayloadTemplate == "" ||
+		service.updateProjectConfig.ResponseTemplate == "" {
+		return project, nil, nil
+	}
+
+	payload, err := generateRequestPayload(project, service.updateProjectConfig.PayloadTemplate)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error generating request payload: %w", err)
+	}
+
+	resp, err := sendUpdateRequest(service.updateProjectConfig.Endpoint, payload)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error sending update request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	response, err := processResponseTemplate(resp, service.updateProjectConfig.ResponseTemplate)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error processing response template: %w", err)
+	}
+
+	return project, response, nil
+}
+
+func generateRequestPayload(project *models.Project, templateString string) (map[string]interface{}, error) {
+	tmpl, err := template.New("requestPayload").Parse(templateString)
+	if err != nil {
+		return nil, err
+	}
+	var payload bytes.Buffer
+	if err := tmpl.Execute(&payload, project); err != nil {
+		return nil, err
+	}
+	var result map[string]interface{}
+	if err := json.Unmarshal(payload.Bytes(), &result); err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func sendUpdateRequest(url string, payload map[string]interface{}) (*http.Response, error) {
+	payloadBytes, err := json.Marshal(payload)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(payloadBytes))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func processResponseTemplate(response *http.Response, templateString string) (map[string]interface{}, error) {
+	var data map[string]interface{}
+	if err := json.NewDecoder(response.Body).Decode(&data); err != nil {
+		return nil, err
+	}
+
+	tmpl, err := template.New("responsePayload").Parse(templateString)
+	if err != nil {
+		return nil, err
+	}
+
+	var responseText bytes.Buffer
+	if err := tmpl.Execute(&responseText, data); err != nil {
+		return nil, err
+	}
+
+	var result map[string]interface{}
+	if err := json.Unmarshal(responseText.Bytes(), &result); err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+// isLabelBlacklisted check if any key in labels is blacklisted
+func isLabelBlacklisted(existingLabels, newLabels []models.Label, blacklist []string) bool {
+	for _, newLabel := range newLabels {
+		if contain(newLabel.Key, blacklist) {
+			for _, existingLabel := range existingLabels {
+				if existingLabel.Key == newLabel.Key && existingLabel.Value != newLabel.Value {
+					return true
+				}
+			}
+		} else {
+			return false
+		}
+	}
+	return false
+}
```

Comment:
You might want to review the entire logic flow of how this function has been implemented... It short circuits unexpectedly in the first iteration of the outer for-loop 🫤 And the function only checks for new labels that happen to be in the blacklist, which is not really what we need to check (what happens if a label that's in the blacklist gets removed from `existingLabels`?).

Here's how you need to think of the blacklist in order to implement it: The blacklist is a list (doesn't have to be a real list/slice, I just mean list in the generic sense) of labels that cannot be modified/removed if they already exist (in the saved project). 

Examples:
```go
blacklist = ["do", "not", "change", "bonus"]
old = {
	"do": "1",
	"not": "2",
	"change": "3",
	"ok": "0"
}

// ALLOWED
ok_1 = {
	"do": "1",
	"not": "2",
	"change": "3",
	"ok": "10", // changed field with label not in blacklist
}

// ALLOWED
ok_2 = {
	"do": "1",
	"not": "2",
	"change": "3",
	"ok": "0",
	"still_ok": "4",  // new field
}

// NOT ALLOWED
block_1 = {
	"do": "2", // changed field with label in blacklist
	"not": "2",
	"change": "3",
	"ok": "0",
}

// NOT ALLOWED
block_2 = {
	// removed field with label in blacklist
	"not": "2",
	"change": "3",
	"ok": "0",
}

// ALLOWED
ok_3 = {
	"do": "1",
	"not": "2",
	"change": "3",
	"ok": "0",
	"bonus": "NEW",  // new field with label in blacklist (edge case)
}
```